tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetDataTest.cpp

```diff
@@ -133,4 +133,103 @@ TEST_F(VersionedLayerClientGetDataTest, GetDataFromPartitionAsync) {
   EXPECT_TRUE(mock_server_client_->Verify());
 }
 
+TEST_F(VersionedLayerClientGetDataTest, GetDataWithHandle) {
+  olp::client::HRN hrn(kTestHrn);
+
+  const auto data_handle =
+      mockserver::ReadDefaultResponses::GenerateDataHandle("test");
+  const auto data = mockserver::ReadDefaultResponses::GenerateData();
+
+  mock_server_client_->MockAuth();
+  mock_server_client_->MockLookupResourceApiResponse(
+      mockserver::ApiDefaultResponses::GenerateResourceApisResponse(
+          kTestHrn));
+
+  mock_server_client_->MockGetResponse(kLayer, data_handle, data);
+
+  auto client = olp::dataservice::read::VersionedLayerClient(
+      hrn, kLayer, boost::none, *settings_);
+
+  auto request = olp::dataservice::read::DataRequest();
+  request.WithDataHandle(data_handle);
+
+  auto future = client.GetData(request);
+  auto data_response = future.GetFuture().get();
+
+  EXPECT_SUCCESS(data_response);
+  ASSERT_TRUE(data_response.GetResult() != nullptr);
+  ASSERT_LT(0, data_response.GetResult()->size());
+  std::string data_string(data_response.GetResult()->begin(),
+                          data_response.GetResult()->end());
+  ASSERT_EQ(data, data_string);
+  EXPECT_TRUE(mock_server_client_->Verify());
+}
+
+TEST_F(VersionedLayerClientGetDataTest, GetDataWithInvalidLayerId) {
+  olp::client::HRN hrn(kTestHrn);
+
+  const auto kInvalidLayer = "InvalidLayer";
+
+  mock_server_client_->MockAuth();
+  mock_server_client_->MockLookupResourceApiResponse(
+      mockserver::ApiDefaultResponses::GenerateResourceApisResponse(
+          kTestHrn));
+  mock_server_client_->MockGetVersionResponse(
+      mockserver::ReadDefaultResponses::GenerateVersionResponse(kVersion));
+  mock_server_client_->MockGetError(
+      olp::client::ApiError(olp::http::HttpStatusCode::BAD_REQUEST),
+      "/query/v1/catalogs/" + std::string(kTestHrn) + "/layers/" +
+          std::string(kInvalidLayer) + "/partitions");
```

Comment:
Maybe, move to some helper function. It used here in 2 tests and I am also going to add test for GetData() that needs such path creation.